src/parsers/manifestjson.js

```diff
@@ -70,6 +70,12 @@ export default class ManifestJSONParser extends JSONParser {
       };
     } else {
       // We've parsed the JSON; now we can validate the manifest.
+      if (
+        this.parsedJSON.browser_specific_settings &&
+        !this.parsedJSON.applications
+      ) {
+        this.parsedJSON.applications = this.parsedJSON.browser_specific_settings;
+      }
```

Comment:
Actually, in Firefox, `browser_specific_settings` takes precedence over `applications` if both are specified.

Reply:
@ExE-Boss is right (thanks for pointing it out!)

@eviljeff here is the relevant part of the implementation on the Firefox side:

```
  let bss = (manifest.browser_specific_settings && manifest.browser_specific_settings.gecko)
      || (manifest.applications && manifest.applications.gecko) || {};
  if (manifest.browser_specific_settings && manifest.applications) {
    logger.warn("Ignoring applications property in manifest");
  }
```

from https://searchfox.org/mozilla-central/rev/55895c49f55073d82d977cb74ec1d3a71ae4b25f/toolkit/mozapps/extensions/internal/XPIInstall.jsm#443-444

Reply:
https://github.com/mozilla/addons-linter/pull/2285/commits/0f73e1805aaca5a9f1dfb1f9d3214a7ff0a922f9